src/utilities/getTabEncryption/getTabEncryption.ts

```diff
@@ -45,34 +52,43 @@ export async function getTabEncryption(
     );
   }
 
-  const encryptionKey = Utils.Crypto.makeEncryptionKeypairFromSeed();
+  const encryptionKey = Crypto.makeEncryptionKeypairFromSeed();
   const { secretKey } = encryptionKey;
 
-  const baseKey = Utils.Crypto.makeKeypairFromSeed(
-    secretKey.slice(0, 32),
-    'sr25519',
-  );
+  const baseKey = Crypto.makeKeypairFromSeed(secretKey.slice(0, 32), 'sr25519');
   const authenticationKey = baseKey.derive(
     '//authentication',
   ) as typeof baseKey;
 
-  const sporranDidDocument = Did.createLightDidDocument({
+  const sporranDidDocument = createLightDidDocument({
     authentication: [authenticationKey],
     keyAgreement: [encryptionKey],
   });
   const sporranEncryptionDidKey = getDidEncryptionKey(sporranDidDocument);
   const sporranEncryptionDidKeyUri =
-    `${sporranDidDocument.uri}${sporranEncryptionDidKey.id}` as DidResourceUri;
+    `${sporranDidDocument.id}${sporranEncryptionDidKey}` as DidUrl;
+
+  const { contentStream } = await DidResolver.dereference(
+    dAppEncryptionKeyUri,
+    {},
+  );
+  const dAppEncryptionDidKey = contentStream as VerificationMethod | undefined;
+
+  if (dAppEncryptionDidKey?.type !== 'Multikey') {
```

Comment:
This new code hints that a convenience method is needed in the SDK

Reply:
I'm not super confident that this is actually needed here. @rflechtner what do you think?

Reply:
@arty-name what would the convenience method do?

Reply:
It would get a key of desired type or throw. Having to do type casting and checking the key type as above is not something to be written every time one gets a key, in my opinion :)

Reply:
a verification method to public key function maybe? Would be great if that was able to turn both the old and the new verification method type into a combination of public key bytes and a key type identifier, such as ed25519 / sr25519 / secp256k1 / x25519 etc.